packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

```diff
@@ -148,3 +158,67 @@ contract OPContractsManager_InternalMethods_Test is Test {
         vm.assertEq(expected, actual);
     }
 }
+
+contract OPContractsManager_Upgrade_Harness is CommonTest {
+    event Upgraded(ISystemConfig indexed systemConfig, address indexed upgrader);
+
+    IProxyAdmin proxyAdmin;
+    address upgrader;
+
+    function setUp() public virtual override {
+        super.disableUpgradedFork();
+        super.setUp();
+        if (!isForkTest()) {
+            // This test is only supported in forked tests, as we are testing the upgrade.
+            vm.skip(true);
+        }
+
+        proxyAdmin = IProxyAdmin(EIP1967Helper.getAdmin(address(systemConfig)));
+        upgrader = proxyAdmin.owner();
+        vm.label(upgrader, "ProxyAdmin Owner");
+    }
+
+    function _getOpChains() internal view returns (OPContractsManager.OpChain[] memory) {
+        OPContractsManager.OpChain[] memory opChains = new OPContractsManager.OpChain[](1);
+        opChains[0] = OPContractsManager.OpChain({ systemConfigProxy: systemConfig, proxyAdmin: proxyAdmin });
+        return opChains;
+    }
+}
+
+contract OPContractsManager_Upgrade_Test is OPContractsManager_Upgrade_Harness {
+    function setUp() public override {
+        // For tests which are specifically testing the upgrade, we want do not want the contracts to have been
+        // upgraded yet.
+        super.disableUpgradedFork();
+        super.setUp();
+    }
+
+    function test_upgrade_succeeds() public {
+        OPContractsManager.OpChain[] memory opChains = _getOpChains();
+        vm.etch(upgrader, vm.getDeployedCode("test/mocks/Callers.sol:DelegateCaller"));
+
+        vm.expectEmit(true, true, true, false);
```

Comment:
we don't have any data, but still best practice to just check everything, in case we add non-indexed data later
```suggestion
        vm.expectEmit();
```

Reply:
fixed